src/movies.js

```diff
@@ -1,28 +1,190 @@
+  let sortedMovies = moviesClone.sort(function(a, b) {
+    return a.title.localeCompare(b.title);
+  });
+
+  // Creates a new array with titles only
+  let onlyNames = sortedMovies.map(function (movie) {
+    return movie.title;
+  });
+  
+  // Returns only the 20 first titles
+  return onlyNames.splice(0,20);
+} 
+// console.log(orderAlphabetically(movies));
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes() {}
+function turnHoursToMinutes(movies) {
+  // Makes a copy so that original array won't be changed
+ let moviesClone = JSON.parse(JSON.stringify(movies));
+ let hours, minutes;
+ 
+ moviesClone.forEach(function(movie) {
+   /* Gets the first character of the duration (for instance 
+   a string with value '2') and parses it to an integer */
+   hours = parseInt(movie.duration.slice(0,1));
+   
+   /* Some movies have exactly 2h duration so they don't have
+   anything on the position 3 of the string. In that case,
+   minutes should receive the value 0.
+   
+   If any other case, the string '22' will become the number 22
+   for instance, and if the minutes are less than 10 (for 
+   example: '2h 3m'), the string will get '3m' but after parsing 
+   to integer, the value will be 3 */
+   minutes = parseInt(movie.duration.slice(3,5)) || 0;
+   movie.duration = hours*60 + minutes;
+ });
+ return moviesClone;
+}
+// console.log (turnHoursToMinutes(movies))
 
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg() {}
+
+function bestYearAvg(movies) {
+  if (movies.length === 0) return null; // In case there are no movies
+
+  // Makes a copy so that original array won't be changed
+  let moviesClone = JSON.parse(JSON.stringify(movies));
+  
+  // Creates a new array with only year and score attributes
+  // of each movie
+  let yearsAndScoresArray = moviesClone.map(function (movie) {
+    return {
+      year: movie.year,
+      score: movie.score
+    };
+  });
+  
+  // Creates a new array with unique years
+  let uniqueYearsArray = [];
+  for (let i = 0; i < yearsAndScoresArray.length; i++) {
+    if (uniqueYearsArray.indexOf(yearsAndScoresArray[i].year) === -1)
+      uniqueYearsArray.push(yearsAndScoresArray[i].year);
+  }
+    
+  /* Creates a new array containing just the sum of scores of every year (stored on the same index 
+  as that of the year)  
+  Also creates yet another array, containing the amount of entries (movies) of that specific year
+  (also on the same index as that of the year) */
+  let uniqueYearsArrayWithRatings = [];
+  let entries = [];
+  for (let i = 0; i < uniqueYearsArray.length; i++) {
+    for (let j = 0; j < yearsAndScoresArray.length; j++) {
+      if (yearsAndScoresArray[j].year === uniqueYearsArray[i]) {
+        uniqueYearsArrayWithRatings[i] = (uniqueYearsArrayWithRatings[i] || 0) + yearsAndScoresArray[j].score;
+        entries[i] = (entries[i] || 0) + 1;
+      }
+    }
+  }
+  
```

Comment:
You are using a lot of helper variable and arrays. As a result, it's a bit hard to read what you are trying to do and it becomes quite long. You can often just map or filter or reduce through the original object (using the dot notation to retrieve the things you need) and then doing something with them. You can also chain methods (arr.map.filter). And you don't need to clone the original array if you're using the methods that return a new array.

I don't really want to give you a solution, Jan already gave one in the review I believe. There a few, as always. I can, but maybe you can try to rewrite it first, make it more efficient and more concise, and then I can give you a couple of solutions?